vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1706 @@
+            }
+        }
+
+        VTR_LOG("Redirects:\n");
+        for (auto& entry : virtual_redirect_) {
+            int node;
+            location loc;
+            std::tie(node, loc) = entry.first;
+
+            location new_loc;
+            e_rr_type new_type;
+            int new_idx;
+            std::tie(new_loc, new_type, new_idx) = entry.second;
+            VTR_LOG("\t Node:%d X:%d Y:%d -> X:%d Y:%d CHAN%c idx:%d\n", node, loc.first, loc.second, new_loc.first,
+                    new_loc.second, new_type == e_rr_type::CHANX ? 'X' : 'Y', new_idx);
+        }
+
+        VTR_LOG("Shorts:\n");
+        for (auto& entry : virtual_shorts_) {
+            location loc1, loc2;
+            e_rr_type type1, type2;
+            int id1, id2;
+            std::tie(loc1, type1, id1, loc2, type2, id2) = entry;
+            VTR_LOG("\tCHAN%c X:%d Y%d", type1 == e_rr_type::CHANX ? 'X' : 'Y', loc1.first, loc1.second);
+            VTR_LOG(" Segment id:%d name:%s ->", id1,
+                    segment_inf_[RRSegmentId(std::get<0>(virtual_chan_loc_map_[std::make_tuple(loc1, type1)][id1]))].name.c_str());
+            VTR_LOG(" CHAN%c X:%d Y:%d", type2 == e_rr_type::CHANX ? 'X' : 'Y', loc2.first, loc2.second);
+            VTR_LOG(" Segment id:%d name:%s\n", id2,
+                    segment_inf_[RRSegmentId(std::get<0>(virtual_chan_loc_map_[std::make_tuple(loc2, type2)][id2]))].name.c_str());
+        }
+    }
+
+    void print() {
+        VTR_LOG("Real\n");
+
+        VTR_LOG("Virtual to real mapping:\n");
+        for (auto i : virtual_beg_to_real_) {
+            e_rr_type type;
+            location loc;
+            int virt_idx;
+            std::tie(loc, type, virt_idx) = i.first;
+            VTR_LOG("CHAN%c X:%d Y:%d virt_idx:%d -> idx:%d\n", type == CHANX ? 'X' : 'Y', loc.first, loc.second, virt_idx, i.second);
+        }
+
+        VTR_LOG("RR_Nodes\n");
+        for (auto i : loc_type_idx_to_rr_idx_) {
+            location loc;
+            e_rr_type type;
+            int idx;
+            std::tie(loc, type, idx) = i.first;
+            VTR_LOG("\t X:%d Y:%d type:%d idx:%d->rr_node:%d\n", loc.first, loc.second, type, idx, i.second);
+        }
+    }
+
+    void fill_switch(t_rr_switch_inf& switch_,
+                     float R,
+                     float Cin,
+                     float Cout,
+                     float Cinternal,
+                     float Tdel,
+                     float mux_trans_size,
+                     float buf_size,
+                     char* name,
+                     SwitchType type) {
+        switch_.R = R;
+        switch_.Cin = Cin;
+        switch_.Cout = Cout;
+        switch_.Cinternal = Cinternal;
+        switch_.Tdel = Tdel;
+        switch_.mux_trans_size = mux_trans_size;
+        switch_.buf_size = buf_size;
+        switch_.name = name;
+        switch_.set_type(type);
+    }
```

Comment:
This function could possibly end up in the `arch_utils.cpp` file as it can be re-used elsewhere

Reply:
Moved to arch_utils.cpp in next commit